### PR TITLE
pretty: miscellaneous formatting improvements

### PR DESCRIPTION
**Pull request by knz**
This is not yet in the domain of "bikeshedding" -- more alike to "general clean-up"

Cumulative perf improvements:

```
name           old time/op    new time/op    delta
PrettyData-16     4.48s ±21%     1.77s ±26%  -60.44%  (p=0.008 n=5+5)
PrettyData-16    3.20GB ± 2%    0.62GB ± 1%  -80.56%  (p=0.008 n=5+5)
PrettyData-16     10.5M ± 0%      4.2M ± 0%  -59.67%  (p=0.008 n=5+5)
```

### CODE REVIEW

**Review by madelynnblue — APPROVED** [review]
These are great. Thanks.

**Comment by knz** [comment]
and there we go

bors r+

File: pkg/util/pretty/document.go [hunk 1]
```diff
@@ -106,6 +106,24 @@ func Concat(a, b Doc) Doc {
 	})
 }
 
+// Fill concatenates a list of docs with spaces if possible,
+// otherwise with newlines.
+// See linked paper pp 14-15.
+// func Fill(d ...Doc) Doc {
```
Comment (mjibson) [inline pkg/util/pretty/document.go]: Was this debug work? Probably wasn't supposed to be in this commit but whatever.

File: pkg/sql/sem/tree/pretty.go [hunk 2]
```diff
@@ -799,7 +799,11 @@ func (node *UpdateExpr) doc(p PrettyCfg) pretty.Doc {
 	if node.Tuple {
 		d = p.bracket("(", d, ")")
 	}
-	return pretty.Stack(d, pretty.Text("="), p.Doc(node.Expr))
+	e := node.Expr
+	if p.Simplify {
+		e = StripParens(e)
+	}
+	return pretty.Group(p.nestName(d, pretty.ConcatSpace(pretty.Text("="), p.Doc(e))))
```
Comment (mjibson) [inline pkg/sql/sem/tree/pretty.go]: nestName calls Group on its outer Doc, so the call to Group here is extraneous.

**Review by madelynnblue — COMMENTED** [review]
I approved this PR and then it changed a bunch, so here are the rest of my comments.

File: pkg/util/pretty/document.go (hunk 1 above)
Comment (knz) [reply pkg/util/pretty/document.go]: I had tried it out but then commented it out to please the linter. I think it's a nice function but we haven't used it. I was happy while writing this to see how easy it was to reproduce the algorithms!

File: pkg/sql/sem/tree/pretty.go (hunk 2 above)
Comment (knz) [reply pkg/sql/sem/tree/pretty.go]: I see you fixed it. Thanks!

File: pkg/sql/sem/tree/pretty.go [hunk 3]
```diff
@@ -252,24 +269,57 @@ func (node *Exprs) doc(p PrettyCfg) pretty.Doc {
...
+func (p PrettyCfg) peelBinaryOperand(e Expr, op BinaryOperator) Expr {
+	if !p.Simplify {
+		return e
+	}
+	stripped := StripParens(e)
+	switch te := stripped.(type) {
+	case *BinaryExpr:
+		parenPrio := binaryOpPrio[op]
+		childPrio := binaryOpPrio[te.Operator]
+		if te.Operator == op && childPrio <= parenPrio {
```
[the remark this answers is not in the document]
Comment (knz) [reply pkg/sql/sem/tree/pretty.go]: Thanks I had broken this inadvertently. See #27255.

**Pull request merged**